planner/core/task.go

```diff
@@ -1077,37 +1142,48 @@ func (p *basePhysicalAgg) newPartialAggregate(copTaskType kv.StoreType) (partial
 	if !CheckAggCanPushCop(p.ctx, p.AggFuncs, p.GroupByItems, copTaskType) {
 		return nil, p.self
 	}
-	finalAggFuncs, finalGbyItems, partialSchema := BuildFinalModeAggregation(p.ctx, p.AggFuncs, p.GroupByItems, p.schema)
+	partialPref, finalPref, funcMap := BuildFinalModeAggregation(p.ctx, &AggPref{
+		AggFuncs:     p.AggFuncs,
+		GroupByItems: p.GroupByItems,
+		Schema:       p.Schema(),
+	})
 	// Remove unnecessary FirstRow.
-	p.AggFuncs = RemoveUnnecessaryFirstRow(p.ctx, finalAggFuncs, finalGbyItems, p.AggFuncs, p.GroupByItems, partialSchema)
+	partialPref.AggFuncs = RemoveUnnecessaryFirstRow(p.ctx,
+		finalPref.AggFuncs, finalPref.GroupByItems,
+		partialPref.AggFuncs, partialPref.GroupByItems, partialPref.Schema, funcMap)
 	if copTaskType == kv.TiDB {
 		// For partial agg of TiDB cop task, since TiDB coprocessor reuse the TiDB executor,
 		// and TiDB aggregation executor won't output the group by value,
 		// so we need add `firstrow` aggregation function to output the group by value.
-		aggFuncs, err := genFirstRowAggForGroupBy(p.ctx, p.GroupByItems)
+		aggFuncs, err := genFirstRowAggForGroupBy(p.ctx, partialPref.GroupByItems)
 		if err != nil {
 			return nil, p.self
 		}
-		p.AggFuncs = append(p.AggFuncs, aggFuncs...)
+		partialPref.AggFuncs = append(partialPref.AggFuncs, aggFuncs...)
 	}
-	finalSchema := p.schema
-	p.schema = partialSchema
+	// finalSchema := p.schema
+	p.AggFuncs = partialPref.AggFuncs
+	p.GroupByItems = partialPref.GroupByItems
+	p.schema = partialPref.Schema
 	partialAgg := p.self
 	// Create physical "final" aggregation.
+	prop := &property.PhysicalProperty{
+		ExpectedCnt: math.MaxFloat64,
+	}
 	if p.tp == plancodec.TypeStreamAgg {
 		finalAgg := basePhysicalAgg{
-			AggFuncs:     finalAggFuncs,
-			GroupByItems: finalGbyItems,
-		}.initForStream(p.ctx, p.stats, p.blockOffset)
-		finalAgg.schema = finalSchema
+			AggFuncs:     finalPref.AggFuncs,
+			GroupByItems: finalPref.GroupByItems,
+		}.initForStream(p.ctx, p.stats, p.blockOffset, prop)
```

Comment:
we should reset the required physical order property for stream aggregations.

Reply:
See https://github.com/pingcap/tidb/pull/15500#discussion_r399380701 and https://github.com/pingcap/tidb/pull/15500#discussion_r399358074